addons/survey/controllers/main.py

```diff
@@ -80,6 +80,14 @@ def _check_validity(self, survey_token, answer_token, ensure_token=True):
         if (not survey_sudo.page_ids and survey_sudo.questions_layout == 'page_per_section') or not survey_sudo.question_ids:
             return 'survey_void'
 
+        if answer_sudo:
+            if request.env.user._is_public():
```

Comment:
As we have early returns, we can gain some indentation just for code readability (and make condition easier to read even if a bit more verbose). I propose some comment update also to be clearer, like
```
if answer_sudo:
    if request.env.user._is_public() and answer_sudo.partner_id:
        # answers from public user should not have any partner_id; this indicates probably a cookie issue
        return 'answer_wrong_user'
    if not request.env.user._is_public() and answer_sudo.partner_id != request.env.user.partner_id:
        # partner mismatch, probably a cookie issue
        return 'answer_wrong_user'
```